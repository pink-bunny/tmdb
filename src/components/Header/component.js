import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import {
  Typography, Row, Col, Avatar, Dropdown, Icon, Menu, Layout
} from 'antd';

const getOverlay = (completeSession) => (
  <Menu>
    <Menu.Item>
      <Link to="/">Dashboard</Link>
    </Menu.Item>
    <Menu.Divider />
    <Menu.Item>
      <Link to="/my-lists">My Lists</Link>
    </Menu.Item>
    <Menu.Item>
      <Link to="/watchlist">Watchlist</Link>
    </Menu.Item>
    <Menu.Item>
      <Link to="/stubs/favorites">Favorites</Link>
    </Menu.Item>
    <Menu.Divider />
    <Menu.Item onClick={completeSession}>Logout</Menu.Item>
  </Menu>
);

const Header = ({ completeSession, userLoggedIn, username }) => (
  <Layout.Header>
    <Row
      type="flex"
      justify="space-between"
    >
      <Col>
        <Typography.Text>THE MOVIE DB</Typography.Text>
      </Col>
      <Col>
        {userLoggedIn && (
          <>
            <Avatar icon="user" />
            {' '}
            <Dropdown overlay={getOverlay(completeSession)}>
              <Typography.Text>
                {`${username} `}
                <Icon type="caret-down" />
              </Typography.Text>
            </Dropdown>
          </>
        )}
      </Col>
    </Row>
  </Layout.Header>
);

Header.propTypes = {
  completeSession: PropTypes.func.isRequired,
  userLoggedIn: PropTypes.bool.isRequired,
  username: PropTypes.string.isRequired
};

export default Header;
