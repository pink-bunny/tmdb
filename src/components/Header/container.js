import { connect } from 'react-redux';

import Header from './component';
import { completeSession } from '../../../state/session/actions';
import { isUserLoggedIn, username } from '../../../state/session/selectors';

const mapDispatchToProps = {
  completeSession
};

const mapStateToProps = (state) => ({
  userLoggedIn: isUserLoggedIn(state),
  username: username(state)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
