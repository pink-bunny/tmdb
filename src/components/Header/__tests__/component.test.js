import React from 'react';
import { shallow } from 'enzyme';

import Header from '../component';

describe('Header component', () => {
  const requiredProps = {
    completeSession: jest.fn(),
    userLoggedIn: false,
    username: ''
  };

  it('matches snapshot when user logged out', () => {
    const wrapper = shallow(<Header {...requiredProps} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });

  it('matches snapshot when user logged in', () => {
    const props = {
      ...requiredProps,
      userLoggedIn: true,
      username: 'John Doe'
    };
    const wrapper = shallow(<Header {...props} />); /* eslint-disable-line */

    expect(wrapper).toMatchSnapshot();
  });
});
