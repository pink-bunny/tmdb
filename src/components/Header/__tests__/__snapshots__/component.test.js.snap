// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot when user logged in 1`] = `
<Adapter>
  <Row
    gutter={0}
    justify="space-between"
    type="flex"
  >
    <Col>
      <Text>
        THE MOVIE DB
      </Text>
    </Col>
    <Col>
      <Avatar
        icon="user"
        shape="circle"
        size="default"
      />
       
      <Dropdown
        mouseEnterDelay={0.15}
        mouseLeaveDelay={0.1}
        overlay={[Function]}
        placement="bottomLeft"
      >
        <Text>
          John Doe 
          <Icon
            type="caret-down"
          />
        </Text>
      </Dropdown>
    </Col>
  </Row>
</Adapter>
`;

exports[`Header component matches snapshot when user logged out 1`] = `
<Adapter>
  <Row
    gutter={0}
    justify="space-between"
    type="flex"
  >
    <Col>
      <Text>
        THE MOVIE DB
      </Text>
    </Col>
    <Col />
  </Row>
</Adapter>
`;
